src/olympia/amo/utils.py

```diff
@@ -1182,10 +1182,12 @@ def is_safe_url(url, request, allowed_hosts=None):
     allowed_hosts = allowed_hosts or (
         settings.DOMAIN,
         urlparse(settings.CODE_MANAGER_URL).netloc,
-        f'{settings.DOMAIN}:{settings.ADDONS_FRONTEND_PROXY_PORT}'
-        if settings.ADDONS_FRONTEND_PROXY_PORT
-        else None,
     )
+    if settings.ADDONS_FRONTEND_PROXY_PORT:
+        allowed_hosts = allowed_hosts + (
+            f'{settings.DOMAIN}:{settings.ADDONS_FRONTEND_PROXY_PORT}',
+        )
```

Comment:
I don't think this is better as it adds the `domain:port` version in every case, even when a custom `allowed_hosts` is passed, which is probably a bit surprising.

Using a separate variable for the "default" allowed hosts or doing `allowed_hosts or (...) + ((f'{settings.DOMAIN}:{settings.ADDONS_FRONTEND_PROXY_PORT}',) if settings.ADDONS_FRONTEND_PROXY_PORT else ())` would fix this. The default variable path is probably going to be a lot easier to read

Reply:
I updated the code to address the issue. I did it a bit differently from how you suggested, but I think it works and is reasonable.